js/modules/k6/experimental/tracing/client.go

```diff
@@ -0,0 +1,237 @@
+type Client struct {
+	vu modules.VU
+
+	// opts holds the client's configuration options.
+	opts options
+
+	// propagator holds the client's trace propagator, used
+	// to produce trace context headers for each supported
+	// formats: w3c, b3, jaeger.
+	propagator Propagator
+
+	// requestFunc holds the http module's request function
+	// used to emit HTTP requests in k6 script. The client
+	// uses it under the hood to emit the requests it
+	// instruments.
+	requestFunc HTTPRequestFunc
+}
+
+// HTTPRequestFunc is a type alias representing the prototype of
+// k6's http module's request function
+type HTTPRequestFunc func(method string, url goja.Value, args ...goja.Value) (*httpmodule.Response, error)
+
+// NewClient instantiates a new tracing Client
+func NewClient(vu modules.VU, opts options) *Client {
+	rt := vu.Runtime()
+
+	// Get the http module's request function
+	httpModuleRequest, err := rt.RunString("require('k6/http').request")
+	if err != nil {
+		common.Throw(
+			rt,
+			fmt.Errorf(
+				"failed initializing tracing client, "+
+					"unable to require http.request method; reason: %w",
+				err,
+			),
+		)
+	}
+
+	// Export the http module's request function goja.Callable as a Go function
+	var requestFunc HTTPRequestFunc
+	if err := rt.ExportTo(httpModuleRequest, &requestFunc); err != nil {
+		common.Throw(
+			rt,
+			fmt.Errorf("failed initializing tracing client, unable to export http.request method; reason: %w", err),
+		)
+	}
+
+	client := &Client{vu: vu, requestFunc: requestFunc}
+	if err := client.Configure(opts); err != nil {
+		common.Throw(
+			rt,
+			fmt.Errorf("failed initializing tracing client, invalid configuration; reason: %w", err),
+		)
+	}
+
+	return client
+}
+
+// Configure configures the tracing client with the given options.
+func (c *Client) Configure(opts options) error {
+	if err := opts.validate(); err != nil {
+		return fmt.Errorf("invalid options: %w", err)
+	}
+
+	switch opts.Propagator {
+	case "w3c":
+		c.propagator = &W3CPropagator{}
+	case "jaeger":
+		c.propagator = &JaegerPropagator{}
+	default:
+		return fmt.Errorf("unknown propagator: %s", opts.Propagator)
+	}
+
+	c.opts = opts
+
+	return nil
+}
+
+// Request instruments the http module's request function with tracing headers,
+// and ensures the trace_id is emitted as part of the output's data points metadata.
+func (c *Client) Request(method string, url goja.Value, args ...goja.Value) (*httpmodule.Response, error) {
+	rt := c.vu.Runtime()
+
+	// The http module's request function expects the first argument to be the
+	// request body. If no body is provided, we need to pass null to the function.
+	if len(args) == 0 {
+		args = []goja.Value{goja.Null()}
+	}
+
+	traceID := TraceID{
+		Prefix: k6Prefix,
+		Code:   k6CloudCode,
+		Time:   time.Now(),
+	}
+
+	encodedTraceID, err := traceID.Encode()
+	if err != nil {
+		common.Throw(rt, fmt.Errorf("failed to encode the generated trace ID; reason: %w", err))
```

Comment:
Given that you here can return an error is there a reason why:
1. you are not returning it
2. don't just not have the err.


I prefer not `common.Throw` where possible as that works better with go code IMO and it is automatically translated. But there are cases where that is either too much work for very little benefit  - not interopping with go code in a wayt that matter.

Reply:
Great catch! This is a leftover from the previous implementation 🤦🏻 I'll fix it in a fixup commit 🙇🏻